account_multicurrency_revaluation/tests/test_currency_revaluation.py

```diff
@@ -0,0 +1,199 @@
+                             'acc_reval_loss').id,
+            'revaluation_gain_account_id':
+                self.env.ref('account_multicurrency_revaluation.'
+                             'acc_reval_gain').id,
+            'default_currency_reval_journal_id': self.reval_journal.id,
+        }
+        company.write(values)
+
+        wizard = self.env['wizard.currency.revaluation']
+        data = {
+            'revaluation_date': '2017-03-15',
+            'journal_id': self.reval_journal.id,
+            'label': '[%(account)s] wiz_test',
+        }
+        wiz = wizard.create(data)
+        result = wiz.revaluate_currency()
+
+        # Assert the wizard show the created revaluation lines
+        self.assertEqual(result.get('name'), "Created revaluation lines")
+
+        reval_move_lines = self.env['account.move.line'].search([
+            ('name', 'like', 'wiz_test')])
+
+        # Assert 8 account.move.line were generated
+        self.assertEqual(len(reval_move_lines), 8)
+
+        for reval_line in reval_move_lines:
+            if reval_line.account_id.name == 'Account Liquidity USD':
+                self.assertFalse(reval_line.partner_id)
+                self.assertEqual(reval_line.credit, 0.0)
+                self.assertEqual(reval_line.debit, 105.0)
+            elif reval_line.account_id.name == 'Account Liquidity GBP':
+                self.assertFalse(reval_line.partner_id)
+                self.assertEqual(reval_line.credit, 0.0)
+                self.assertEqual(reval_line.debit, 105.0)
+            elif reval_line.account_id.name == 'Account Receivable':
+                self.assertIsNotNone(reval_line.partner_id.id)
+                self.assertEqual(reval_line.credit, 185.0)
+                self.assertEqual(reval_line.debit, 0.0)
+            elif reval_line.account_id.name == 'Reval Gain':
+                self.assertEqual(reval_line.credit, 105.0)
+                self.assertEqual(reval_line.debit, 0.0)
+            elif reval_line.account_id.name == 'Reval Loss':
+                self.assertEqual(reval_line.credit, 0.0)
+                self.assertEqual(reval_line.debit, 185.0)
+
+    def test_defaults(self):
+        self.env['res.config.settings'].create({
+            'default_currency_reval_journal_id': self.reval_journal.id,
+            'revaluation_loss_account_id':
+                self.env.ref('account_multicurrency_revaluation.'
+                             'acc_reval_loss').id,
+            'revaluation_gain_account_id':
+                self.env.ref('account_multicurrency_revaluation.'
+                             'acc_reval_gain').id
+        }).execute()
+
+        wizard = self.env['wizard.currency.revaluation'].create({})
+
+        self.assertEqual(wizard.revaluation_date,
+                         fields.date.today().strftime(DATE_FORMAT))
+        self.assertEqual(wizard.journal_id, self.reval_journal)
+
+    def setUp(self):
```

Comment:
better to:
1. inherit from SavepointCase
2. move setUp code to setUpClass
3. move this method at the top